test/app/features/profile/domain/usecases/user_profile_test.dart

```diff
@@ -1,90 +1,115 @@
-import '../../../../../utils/helper.mocks.dart';
+class MockAppStateUseCase extends Mock implements AppStateUseCase {}
+
+class MockUserProfileRepository extends Mock implements IUserProfileRepository {
+}
+
+class MockUserProfileStore extends Mock
+    implements LocalStore<UserProfileEntity> {}
 
 void main() {
-  late final MockAppStateUseCase appStateUseCase = MockAppStateUseCase();
-  late final MockIUserProfileRepository repository =
-      MockIUserProfileRepository();
-  late final MockUserProfileStore profileStore = MockUserProfileStore();
+  late AppStateUseCase appStateUseCase;
+  late IUserProfileRepository repository;
+  late LocalStore<UserProfileEntity> profileStore;
+  late UserProfile sut;
 
-  late final UserProfile sut = UserProfile(
-    repository: repository,
-    userProfileStore: profileStore,
-    appStateUseCase: appStateUseCase,
-  );
+  setUp(
+    () {
+      appStateUseCase = MockAppStateUseCase();
+      repository = MockUserProfileRepository();
+      profileStore = MockUserProfileStore();
 
-  setUp(() {
-    when(appStateUseCase.check()).thenAnswer(
-      (_) => Future.value(
-        right(
+      sut = UserProfile(
+        repository: repository,
+        userProfileStore: profileStore,
+        appStateUseCase: appStateUseCase,
+      );
+
+      when(() => appStateUseCase.check()).thenAnswer(
+        (_) async => right(
           const AppStateEntity(
             quizSession: null,
             userProfile: null,
             appMode: AppStateModeEntity(),
             modules: [],
           ),
         ),
-      ),
-    );
-  });
-
-  group('UserProfile', () {
-    test('should enable stealth mode', () async {
-      // arrange
-      final actual = right(const ValidField());
-      when(repository.stealthMode(toggle: anyNamed('toggle')))
-          .thenAnswer((_) async => right(const ValidField()));
+      );
+    },
+  );
 
-      // act
-      final expected = await sut.stealthMode(toggle: true);
-      // assert
-      expect(actual, expected);
-      verify(repository.stealthMode(toggle: true));
-      verify(appStateUseCase.check());
-    });
+  group(UserProfile, () {
+    test(
+      'enable stealth mode',
+      () async {
+        // arrange
+        final actual = right(const ValidField());
+        when(() => repository.stealthMode(toggle: any(named: 'toggle')))
+            .thenAnswer((_) async => right(const ValidField()));
 
-    test('should disable stealth mode', () async {
-      // arrange
-      final actual = right(const ValidField());
-      when(repository.stealthMode(toggle: anyNamed('toggle')))
-          .thenAnswer((_) async => right(const ValidField()));
-      // act
-      final expected = await sut.stealthMode(toggle: false);
-      // assert
-      expect(actual, expected);
-      verify(repository.stealthMode(toggle: false));
-    });
+        // act
+        final expected = await sut.stealthMode(toggle: true);
+        // assert
+        expect(actual, expected);
```

Comment:
Sei que foi mantida a estrutura dos testes, porém poderia aproveitar e ajustar o nome das variáveis? Pois caso o teste falhe, a mensagem pode ficar um pouco confuso de se entender sendo que apareceria que o retorno de `stealthMode` (que estaria errado) é o esperado.

```suggestion
        final expected = right(const ValidField());
        when(() => repository.stealthMode(toggle: any(named: 'toggle')))
            .thenAnswer((_) async => right(const ValidField()));

        // act
        final actual = await sut.stealthMode(toggle: true);
        // assert
        expect(actual, expected);
```